src/ipc/ipc4/handler.c

```diff
@@ -215,120 +233,79 @@ static bool is_any_ppl_active(void)
  *     ERROR Stop       EOS       |______\ SAVE
  *                                      /
  */
-int set_pipeline_state(struct ipc_comp_dev *ppl_icd, uint32_t cmd,
-		       bool *delayed)
+
+static int ipc4_pipeline_prepare(struct ipc_comp_dev *ppl_icd, uint32_t cmd)
 {
 	struct ipc_comp_dev *host = NULL;
 	struct ipc *ipc = ipc_get();
 	int status;
-	int ret;
+	int ret = 0;
 
 	status = ppl_icd->pipeline->status;
-	tr_dbg(&ipc_tr, "ipc4 set pipeline %d cmd state %x: from state %x", ppl_icd->id,
-	       cmd, status);
-
-	/* source & sink components are set when pipeline is set to COMP_STATE_INIT */
-	if (status != COMP_STATE_INIT) {
```

Comment:
If I understand correctly removing this block of code is a change of behaviour. At the *very* least this should be mentioned in the commit message. Ideally we could split this commit into two - one just splitting the function with no functional changes, and one with this change. I wouldn't insist on that, but at least let's add a comment to make sure everybody's aware of this change.